ui/app/components/gutter-menu.js

```diff
@@ -37,7 +37,13 @@ export default Component.extend({
   gotoJobsForNamespace(namespace) {
     if (!namespace || !namespace.get('id')) return;
 
-    this.router.transitionTo('jobs', {
+    // Jobs and CSI Volumes are both namespace-sensitive. Changing namespaces is
+    // an intent to reset context, but where to reset to depends on where the namespace
+    // is being switched from. Jobs take precedence, but if the namespace is switched from
+    // a storage-related page, context should be reset to volumes.
+    const destination = this.router.currentRouteName.startsWith('csi.') ? 'csi.volumes' : 'jobs';
+
+    this.router.transitionTo(destination, {
```

Comment:
This was already merged, right?

Reply:
Yes. I must have missed a spot when doing git surgery.